### ISSUE TITLE
Dynamic DCC ports

### ISSUE THREAD

**Issue by SoniEx2** [pircbotx/pircbotx#282]
Currently DCC ports are static, there's no handler for DCC ports, and they're all set in the config.

There should be a way to set a "handler" for DCC ports, with an API like:

```
int openDccPort(); // returns -1, or alternatively throws an exception when it fails to open the port
void closeDccPort(int port);
```
#279 would appreciate this, as you wouldn't keep ports reserved all the time, instead allocating the ports as needed.

**Comment by TheLQ** [comment]
Yes a generic interface would help, then we could have an implementation that uses upnp, a static list of ports, or 0 and lets the OS pick one.
